tools/cli/util.go

```diff
@@ -235,24 +237,58 @@ func anyToString(d interface{}) string {
 	}
 }
 
-func valueToString(v reflect.Value) string {
+func valueToString(v reflect.Value, printFully bool) string {
 	switch v.Kind() {
 	case reflect.Ptr:
-		return valueToString(v.Elem())
+		return valueToString(v.Elem(), printFully)
 	case reflect.Struct:
-		return anyToString(v.Interface())
+		return anyToString(v.Interface(), printFully)
 	case reflect.Invalid:
 		return ""
 	case reflect.Slice:
 		if v.Type().Elem().Kind() == reflect.Uint8 {
-			return fmt.Sprintf("[%v]", string(v.Bytes()))
+			n := string(v.Bytes())
+			if n != "" && n[len(n)-1] == '\n' {
+				return fmt.Sprintf("[%v]", n[:len(n)-1])
+			}
+			return fmt.Sprintf("[%v]", n)
 		}
 		return fmt.Sprintf("[len=%d]", v.Len())
 	default:
 		return fmt.Sprint(v.Interface())
 	}
 }
 
+// limit the maximum length for each field
+func trimInput(input string) string {
```

Comment:
Can you pick a better name than "input". This confuses me a while. Looks like input is one of the field value that may need to be trimmed. But it can be generic to any filed value.

Reply:
Changed